strumpack@7.0.1 +rocm %cce@14.0.2: ld.lld: error: undefined symbol: omp_get_max_threads

### Steps to reproduce the issue

`strumpack@7.0.1 ~slate +rocm amdgpu_target=gfx90a` fails to build using:
* OLCF Crusher, PrgEnv-cray
* `spack@e4s-22.11` (basically `spack@develop`, 008041dea30e99695623f8a742ad7e8f8921e7c5 from November 8, 2022)
* ROCm 5.2.0, registered as external with Spack via `packages:`

Spack environment: [spack.yaml.txt](https://github.com/spack/spack/files/10481903/spack.yaml.txt)

<details><summary>Concretization</summary><pre>
 -   x4mqvon  strumpack@7.0.1%cce@14.0.2+butterflypack+c_interface~count_flops~cuda~ipo+mpi+openmp+parmetis+rocm~scotch+shared~slate~task_timers+zfp amdgpu_target=gfx90a build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen3
[+]  ndl4nqi      ^butterflypack@2.2.2%cce@14.0.2~ipo+openmp+shared build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen3
[+]  vq7kd6m          ^arpack-ng@3.8.0%cce@14.0.2~ipo+mpi+shared build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen3
[+]  aigadm2          ^sed@4.8%cce@14.0.2 build_system=autotools arch=linux-sles15-zen3
[+]  hndeq6h      ^cmake@3.24.2%cce@14.0.2~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-sles15-zen3
[+]  i6ysusq          ^ncurses@6.3%cce@14.0.2~symlinks+termlib abi=none build_system=autotools arch=linux-sles15-zen3
[+]  hm6s4rq              ^pkgconf@1.8.0%cce@14.0.2 build_system=autotools arch=linux-sles15-zen3
[+]  ckjfgdt          ^openssl@1.1.1s%cce@14.0.2~docs~shared build_system=generic certs=mozilla arch=linux-sles15-zen3
[+]  7jk42pl              ^ca-certificates-mozilla@2022-10-11%cce@14.0.2 build_system=generic arch=linux-sles15-zen3
[+]  ljhlvw5              ^zlib@1.2.13%cce@14.0.2+optimize+pic+shared build_system=makefile arch=linux-sles15-zen3
[+]  toktlgw      ^cray-mpich@8.1.17%cce@14.0.2+wrappers build_system=generic arch=linux-sles15-zen3
[+]  2vzy2oi      ^hip@5.2.0%cce@14.0.2~ipo build_system=cmake build_type=Release patches=959d1fe arch=linux-sles15-zen3
[+]  cu6girh      ^hipblas@5.2.0%cce@14.0.2~ipo build_system=cmake build_type=Release arch=linux-sles15-zen3
[+]  azrgbbe      ^hipsparse@5.2.0%cce@14.0.2~ipo build_system=cmake build_type=Release arch=linux-sles15-zen3
[+]  exzk64u      ^hsa-rocr-dev@5.2.0%cce@14.0.2+image~ipo+shared build_system=cmake build_type=Release patches=71e6851 arch=linux-sles15-zen3
[+]  2j6kske      ^llvm-amdgpu@5.2.0%cce@14.0.2~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release patches=a08bbe1 arch=linux-sles15-zen3
[+]  hn2mdex      ^metis@5.1.0%cce@14.0.2~gdb~int64~ipo~real64+shared build_system=cmake build_type=RelWithDebInfo patches=4991da9,93a7903 arch=linux-sles15-zen3
[+]  rs5yuq7      ^netlib-scalapack@2.2.0%cce@14.0.2~ipo~pic+shared build_system=cmake build_type=Release patches=072b006,1c9ce5f arch=linux-sles15-zen3
[+]  ptzsx7h      ^openblas@0.3.20%cce@14.0.2~bignuma~consistent_fpcsr~ilp64+locking+pic+shared build_system=makefile patches=9f12903,f432226 symbol_suffix=none threads=openmp arch=linux-sles15-zen3
[+]  iun7t3b          ^perl@5.36.0%cce@14.0.2+cpanm+shared+threads build_system=generic arch=linux-sles15-zen3
[+]  qks6dq6              ^berkeley-db@18.1.40%cce@14.0.2+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-sles15-zen3
[+]  6xx2ucn              ^bzip2@1.0.8%cce@14.0.2~debug~pic+shared build_system=generic arch=linux-sles15-zen3
[+]  xqsvoxr                  ^diffutils@3.8%cce@14.0.2 build_system=autotools arch=linux-sles15-zen3
[+]  7exjcik                      ^libiconv@1.16%cce@14.0.2 build_system=autotools libs=shared,static arch=linux-sles15-zen3
[+]  vibabgn              ^gdbm@1.23%cce@14.0.2 build_system=autotools arch=linux-sles15-zen3
[+]  mh4vlks                  ^readline@8.1.2%cce@14.0.2 build_system=autotools arch=linux-sles15-zen3
[+]  qrtofw2      ^parmetis@4.0.3%cce@14.0.2~gdb~int64~ipo+shared build_system=cmake build_type=RelWithDebInfo patches=4f89253,50ed208,704b84f arch=linux-sles15-zen3
[+]  jr56bq2      ^rocsolver@5.2.0%cce@14.0.2~ipo+optimal amdgpu_target=auto build_system=cmake build_type=Release arch=linux-sles15-zen3
[+]  z7htnpi      ^zfp@0.5.5%cce@14.0.2~aligned~c~cuda~fasthash~fortran~ipo~openmp~profile~python+shared~strided~twoway bsws=64 build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen3
</pre></details>

Error:
```
$> spack -e . install
...
==> Installing strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm
==> No binary for strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/dd/ddbf9c0509eaf0f8a4c70f59508787336a05eeacc8322f156117d8ce59a70a60.tar.gz
==> No patches needed for strumpack
==> strumpack: Executing phase: 'cmake'
==> strumpack: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

45 errors found in build log:
     538    [100%] Linking CXX executable test_sparse_mpi
     539    cd /tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon/test && /gpfs/alpine/csc439/world-shared/E4S/ParaTools/2
            2.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/cmake-3.24.2-hndeq6hbs2fmlujzbvegtinoxkz2evug/bin/cmake -E cmake_link_script CMakeFiles/test_sparse_mpi.
            dir/link.txt --verbose=1
     540    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/lib/spack/env/cce/case-insensitive/crayCC -O2 -g -DNDEBUG CMakeFiles/test_sparse_mpi.dir/test_s
            parse_mpi.cpp.o -o test_sparse_mpi  -Wl,-rpath,/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon:/gpfs/alpine
            /csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qrtofw2ukuasnwfdugssgoj63whgqswy/lib:/gpfs/alpine/csc
            439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-5.1.0-hn2mdexwfceriurodwmlvpow2kudhzt2/lib:/gpfs/alpine/csc439/wor
            ld-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/zfp-0.5.5-z7htnpirt3x377cljq54qdnx4aphipaj/lib64:/gpfs/alpine/csc439/world-shar
            ed/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64:/gpfs/alpine/csc439/world-s
            hared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/netlib-scalapack-2.2.0-rs5yuq73tfurivq7vvhsvn32y5yijrnp/lib:/gpfs/alpine/csc439/wor
            ld-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/openblas-0.3.20-ptzsx7hwmgk64qtsum43z5rwj6max4qa/lib:/gpfs/alpine/csc439/world-
            shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/arpack-ng-3.8.0-vq7kd6ma24bnp3wtqexjo34sh6jwkup6/lib64:/opt/rocm-5.2.0/lib ../libs
            trumpack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qrtofw2ukuasnwfdugssgoj63whgq
            swy/lib/libparmetis.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-5.1.0-hn2mdexwfceriurodwmlv
            pow2kudhzt2/lib/libmetis.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/zfp-0.5.5-z7htnpirt3x377cljq
            54qdnx4aphipaj/lib64/libzfp.so.0.5.5 /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2
            -ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libsbutterflypack.so.2.2.1 /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/
            cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libdbutterflypack.so.2.2.1 /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spac
            k/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libcbutterflypack.so.2.2.1 /gpfs/alpine/csc439/world-shared/E4S/Pa
            raTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libzbutterflypack.so.2.2.1 /gpfs/alp
            ine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/netlib-scalapack-2.2.0-rs5yuq73tfurivq7vvhsvn32y5yijrnp/lib/libsc
            alapack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/openblas-0.3.20-ptzsx7hwmgk64qtsum43z5rwj6max
            4qa/lib/libopenblas.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/arpack-ng-3.8.0-vq7kd6ma24bnp3wtq
            exjo34sh6jwkup6/lib64/libparpack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/arpack-ng-3.8.0-vq7k
            d6ma24bnp3wtqexjo34sh6jwkup6/lib64/libarpack.so -lm /opt/rocm-5.2.0/lib/libhipblas.so.0.1.50200 /opt/rocm-5.2.0/lib/librocsolver.so.0.1.50200 /opt/rocm-5.2.0/lib/libr
            ocblas.so.0.1.50200 /opt/rocm-5.2.0/lib/libamdhip64.so.5.2.50200 /opt/rocm-5.2.0/llvm/lib/clang/14.0.0/lib/linux/libclang_rt.builtins-x86_64.a /opt/rocm-5.2.0/lib/lib
            hipsparse.so.0.1.50200
     541    [100%] Linking CXX executable test_structure_reuse_mpi
     542    cd /tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon/test && /gpfs/alpine/csc439/world-shared/E4S/ParaTools/2
            2.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/cmake-3.24.2-hndeq6hbs2fmlujzbvegtinoxkz2evug/bin/cmake -E cmake_link_script CMakeFiles/test_structure_r
            euse_mpi.dir/link.txt --verbose=1
     543    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/lib/spack/env/cce/case-insensitive/crayCC -O2 -g -DNDEBUG CMakeFiles/test_structure_reuse_mpi.d
            ir/test_structure_reuse_mpi.cpp.o -o test_structure_reuse_mpi  -Wl,-rpath,/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack
            -build-x4mqvon:/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qrtofw2ukuasnwfdugssgoj63w
            hgqswy/lib:/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-5.1.0-hn2mdexwfceriurodwmlvpow2kudhzt2/
            lib:/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/zfp-0.5.5-z7htnpirt3x377cljq54qdnx4aphipaj/lib64:/gp
            fs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64:
            /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/netlib-scalapack-2.2.0-rs5yuq73tfurivq7vvhsvn32y5yijrnp/
            lib:/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/openblas-0.3.20-ptzsx7hwmgk64qtsum43z5rwj6max4qa/lib
            :/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/arpack-ng-3.8.0-vq7kd6ma24bnp3wtqexjo34sh6jwkup6/lib64:
            /opt/rocm-5.2.0/lib ../libstrumpack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qr
            tofw2ukuasnwfdugssgoj63whgqswy/lib/libparmetis.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-
            5.1.0-hn2mdexwfceriurodwmlvpow2kudhzt2/lib/libmetis.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/z
            fp-0.5.5-z7htnpirt3x377cljq54qdnx4aphipaj/lib64/libzfp.so.0.5.5 /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce
            -14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libsbutterflypack.so.2.2.1 /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/o
            pt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libdbutterflypack.so.2.2.1 /gpfs/alpine/csc439/world-shared/E4S/ParaT
            ools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libcbutterflypack.so.2.2.1 /gpfs/alpine
            /csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/lib64/libzbutte
            rflypack.so.2.2.1 /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/netlib-scalapack-2.2.0-rs5yuq73tfurivq
            7vvhsvn32y5yijrnp/lib/libscalapack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/openblas-0.3.20-pt
            zsx7hwmgk64qtsum43z5rwj6max4qa/lib/libopenblas.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/arpack
            -ng-3.8.0-vq7kd6ma24bnp3wtqexjo34sh6jwkup6/lib64/libparpack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-
            14.0.2/arpack-ng-3.8.0-vq7kd6ma24bnp3wtqexjo34sh6jwkup6/lib64/libarpack.so -lm /opt/rocm-5.2.0/lib/libhipblas.so.0.1.50200 /opt/rocm-5.2.0/lib/librocsolver.so.0.1.502
            00 /opt/rocm-5.2.0/lib/librocblas.so.0.1.50200 /opt/rocm-5.2.0/lib/libamdhip64.so.5.2.50200 /opt/rocm-5.2.0/llvm/lib/clang/14.0.0/lib/linux/libclang_rt.builtins-x86_6
            4.a /opt/rocm-5.2.0/lib/libhipsparse.so.0.1.50200
  >> 544    ld.lld: error: undefined symbol: omp_get_max_threads
     545    >>> referenced by test_BLR_mpi.cpp:86 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_BLR_mpi.cpp:86)
     546    >>>               CMakeFiles/test_BLR_mpi.dir/test_BLR_mpi.cpp.o:(main)
  >> 547    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 548    make[2]: *** [test/CMakeFiles/test_BLR_mpi.dir/build.make:118: test/test_BLR_mpi] Error 1
     549    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 550    make[1]: *** [CMakeFiles/Makefile2:1417: test/CMakeFiles/test_BLR_mpi.dir/all] Error 2
     551    make[1]: *** Waiting for unfinished jobs....
  >> 552    ld.lld: error: undefined symbol: omp_get_max_threads
     553    >>> referenced by test_BLR_seq.cpp:206 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_BLR_seq.cpp:206)
     554    >>>               CMakeFiles/test_BLR_seq.dir/test_BLR_seq.cpp.o:(main)
     555
  >> 556    ld.lld: error: undefined symbol: _cray$mt_kmpc_fork_call_with_flags
     557    >>> referenced by test_BLR_seq.cpp:212 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_BLR_seq.cpp:212)
     558    >>>               CMakeFiles/test_BLR_seq.dir/test_BLR_seq.cpp.o:(main)
     559
  >> 560    ld.lld: error: undefined symbol: __kmpc_single
     561    >>> referenced by test_BLR_seq.cpp:213 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_BLR_seq.cpp:213)
     562    >>>               CMakeFiles/test_BLR_seq.dir/test_BLR_seq.cpp.o:(main_cray$mt$p0003)
     563
  >> 564    ld.lld: error: undefined symbol: __kmpc_end_single
     565    >>> referenced by test_BLR_seq.cpp:214 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_BLR_seq.cpp:214)
     566    >>>               CMakeFiles/test_BLR_seq.dir/test_BLR_seq.cpp.o:(main_cray$mt$p0003)
     567    >>> referenced by test_BLR_seq.cpp:214 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_BLR_seq.cpp:214)
     568    >>>               CMakeFiles/test_BLR_seq.dir/test_BLR_seq.cpp.o:(main_cray$mt$p0003)
  >> 569    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 570    make[2]: *** [test/CMakeFiles/test_BLR_seq.dir/build.make:118: test/test_BLR_seq] Error 1
     571    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 572    make[1]: *** [CMakeFiles/Makefile2:1287: test/CMakeFiles/test_BLR_seq.dir/all] Error 2
  >> 573    ld.lld: error: undefined symbol: omp_get_max_threads
     574    >>> referenced by test_matrix_IO.cpp:187 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_matrix_IO.cpp:187
            )
     575    >>>               CMakeFiles/test_matrix_IO.dir/test_matrix_IO.cpp.o:(main)
     576
  >> 577    ld.lld: error: undefined symbol: _cray$mt_kmpc_fork_call_with_flags
     578    >>> referenced by test_matrix_IO.cpp:193 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_matrix_IO.cpp:193
            )
     579    >>>               CMakeFiles/test_matrix_IO.dir/test_matrix_IO.cpp.o:(main)
     580
  >> 581    ld.lld: error: undefined symbol: __kmpc_single
     582    >>> referenced by test_matrix_IO.cpp:194 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_matrix_IO.cpp:194
            )
     583    >>>               CMakeFiles/test_matrix_IO.dir/test_matrix_IO.cpp.o:(main_cray$mt$p0003)
     584
  >> 585    ld.lld: error: undefined symbol: __kmpc_end_single
     586    >>> referenced by test_matrix_IO.cpp:195 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_matrix_IO.cpp:195
            )
     587    >>>               CMakeFiles/test_matrix_IO.dir/test_matrix_IO.cpp.o:(main_cray$mt$p0003)
     588    >>> referenced by test_matrix_IO.cpp:195 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_matrix_IO.cpp:195
            )
     589    >>>               CMakeFiles/test_matrix_IO.dir/test_matrix_IO.cpp.o:(main_cray$mt$p0003)
  >> 590    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 591    make[2]: *** [test/CMakeFiles/test_matrix_IO.dir/build.make:118: test/test_matrix_IO] Error 1
     592    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 593    make[1]: *** [CMakeFiles/Makefile2:1313: test/CMakeFiles/test_matrix_IO.dir/all] Error 2
  >> 594    ld.lld: error: undefined symbol: omp_get_max_threads
     595    >>> referenced by test_sparse_seq.cpp:110 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_sparse_seq.cpp:1
            10)
     596    >>>               CMakeFiles/test_sparse_seq.dir/test_sparse_seq.cpp.o:(main)
  >> 597    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 598    make[2]: *** [test/CMakeFiles/test_sparse_seq.dir/build.make:118: test/test_sparse_seq] Error 1
     599    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 600    make[1]: *** [CMakeFiles/Makefile2:1261: test/CMakeFiles/test_sparse_seq.dir/all] Error 2
  >> 601    ld.lld: error: undefined symbol: omp_get_max_threads
     602    >>> referenced by test_HSS_seq.cpp:270 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_seq.cpp:270)
     603    >>>               CMakeFiles/test_HSS_seq.dir/test_HSS_seq.cpp.o:(main)
     604
  >> 605    ld.lld: error: undefined symbol: _cray$mt_kmpc_fork_call_with_flags
     606    >>> referenced by test_HSS_seq.cpp:276 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_seq.cpp:276)
     607    >>>               CMakeFiles/test_HSS_seq.dir/test_HSS_seq.cpp.o:(main)
     608
  >> 609    ld.lld: error: undefined symbol: __kmpc_single
     610    >>> referenced by test_HSS_seq.cpp:277 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_seq.cpp:277)
     611    >>>               CMakeFiles/test_HSS_seq.dir/test_HSS_seq.cpp.o:(main_cray$mt$p0003)
     612
  >> 613    ld.lld: error: undefined symbol: __kmpc_end_single
     614    >>> referenced by test_HSS_seq.cpp:278 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_seq.cpp:278)
     615    >>>               CMakeFiles/test_HSS_seq.dir/test_HSS_seq.cpp.o:(main_cray$mt$p0003)
     616    >>> referenced by test_HSS_seq.cpp:278 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_seq.cpp:278)
     617    >>>               CMakeFiles/test_HSS_seq.dir/test_HSS_seq.cpp.o:(main_cray$mt$p0003)
  >> 618    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 619    make[2]: *** [test/CMakeFiles/test_HSS_seq.dir/build.make:118: test/test_HSS_seq] Error 1
     620    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 621    make[1]: *** [CMakeFiles/Makefile2:1235: test/CMakeFiles/test_HSS_seq.dir/all] Error 2
  >> 622    ld.lld: error: undefined symbol: omp_get_max_threads
     623    >>> referenced by test_HSS_mpi.cpp:303 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_mpi.cpp:303)
     624    >>>               CMakeFiles/test_HSS_mpi.dir/test_HSS_mpi.cpp.o:(main)
     625
  >> 626    ld.lld: error: undefined symbol: _cray$mt_kmpc_fork_call_with_flags
     627    >>> referenced by test_HSS_mpi.cpp:313 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_mpi.cpp:313)
     628    >>>               CMakeFiles/test_HSS_mpi.dir/test_HSS_mpi.cpp.o:(main)
     629
  >> 630    ld.lld: error: undefined symbol: __kmpc_single
     631    >>> referenced by test_HSS_mpi.cpp:314 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_mpi.cpp:314)
     632    >>>               CMakeFiles/test_HSS_mpi.dir/test_HSS_mpi.cpp.o:(main_cray$mt$p0003)
     633
  >> 634    ld.lld: error: undefined symbol: __kmpc_end_single
     635    >>> referenced by test_HSS_mpi.cpp:315 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_mpi.cpp:315)
     636    >>>               CMakeFiles/test_HSS_mpi.dir/test_HSS_mpi.cpp.o:(main_cray$mt$p0003)
     637    >>> referenced by test_HSS_mpi.cpp:315 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_HSS_mpi.cpp:315)
     638    >>>               CMakeFiles/test_HSS_mpi.dir/test_HSS_mpi.cpp.o:(main_cray$mt$p0003)
  >> 639    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 640    make[2]: *** [test/CMakeFiles/test_HSS_mpi.dir/build.make:118: test/test_HSS_mpi] Error 1
     641    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 642    make[1]: *** [CMakeFiles/Makefile2:1339: test/CMakeFiles/test_HSS_mpi.dir/all] Error 2
  >> 643    ld.lld: error: undefined symbol: omp_get_max_threads
     644    >>> referenced by test_sparse_mpi.cpp:140 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_sparse_mpi.cpp:1
            40)
     645    >>>               CMakeFiles/test_sparse_mpi.dir/test_sparse_mpi.cpp.o:(main)
  >> 646    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 647    make[2]: *** [test/CMakeFiles/test_sparse_mpi.dir/build.make:118: test/test_sparse_mpi] Error 1
     648    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 649    make[1]: *** [CMakeFiles/Makefile2:1365: test/CMakeFiles/test_sparse_mpi.dir/all] Error 2
  >> 650    ld.lld: error: undefined symbol: omp_get_max_threads
     651    >>> referenced by test_structure_reuse_mpi.cpp:169 (/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-src/test/test_structu
            re_reuse_mpi.cpp:169)
     652    >>>               CMakeFiles/test_structure_reuse_mpi.dir/test_structure_reuse_mpi.cpp.o:(main)
  >> 653    clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 654    make[2]: *** [test/CMakeFiles/test_structure_reuse_mpi.dir/build.make:118: test/test_structure_reuse_mpi] Error 1
     655    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 656    make[1]: *** [CMakeFiles/Makefile2:1391: test/CMakeFiles/test_structure_reuse_mpi.dir/all] Error 2
     657    make[1]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-7.0.1-x4mqvonbxvcad55smbm6efk5os76xuxm/spack-build-x4mqvon'
  >> 658    make: *** [Makefile:139: all] Error 2
```

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (32913973c36dfc6ea465db3f4ecea7689f49ea9f)
* **Python:** 3.9.7
* **Platform:** linux-sles15-zen3
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/10481867/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/10481868/spack-build-out.txt)
[CMakeError.log](https://github.com/spack/spack/files/10481882/CMakeError.log)
[CMakeOutput.log](https://github.com/spack/spack/files/10481883/CMakeOutput.log)

@pghysels @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

> Can you try strumpack@master? Also, I see it is using /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/lib/spack/env/cce/case-insensitive/crayCC for the C++ compiler. Can you try hipcc instead?

Trying `stumpack@master` with PrgEnv-cray:
```
==> Installing strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs
==> No binary for strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs found: installing from source
==> No patches needed for strumpack
==> strumpack: Executing phase: 'cmake'
==> strumpack: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

28 errors found in build log:
     174    [ 11%] Building CXX object CMakeFiles/strumpack.dir/src/misc/Triplet.cpp.o
     175    [ 12%] Building CXX object CMakeFiles/strumpack.dir/src/sparse/CompressedSparseMatrix.cpp.o
     176    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/lib/spack/env/cce/case-insensitive/crayCC -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -Dstrump
            ack_EXPORTS -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-maste
            r-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-build-mp5kr4f -isystem /opt/rocm-5.2.0/include -isystem /opt/rocm-5.2.0/llvm/lib/clang/14.0.0/.. -isystem /gpfs/alpine/csc439/worl
            d-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qrtofw2ukuasnwfdugssgoj63whgqswy/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-5.1.0-hn2mdexwfceriurodwmlvpow2kudhzt2/include -isystem /gpfs/alpine/csc439
            /world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/zfp-0.5.5-z7htnpirt3x377cljq54qdnx4aphipaj/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/include -O2 -g -DNDEBUG -fPI
            C -Wall -Wno-overloaded-virtual -x hip --offload-arch=gfx900 --offload-arch=gfx906 --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx1030 -fopenmp -std=c++17 -
            MD -MT CMakeFiles/strumpack.dir/src/sparse/MC64ad.cpp.o -MF CMakeFiles/strumpack.dir/src/sparse/MC64ad.cpp.o.d -o CMakeFiles/strumpack.dir/src/sparse/MC64ad.cpp.o -c /tmp/
            lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src/sparse/MC64ad.cpp
     177    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/lib/spack/env/cce/case-insensitive/crayCC -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -Dstrump
            ack_EXPORTS -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-maste
            r-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-build-mp5kr4f -isystem /opt/rocm-5.2.0/include -isystem /opt/rocm-5.2.0/llvm/lib/clang/14.0.0/.. -isystem /gpfs/alpine/csc439/worl
            d-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qrtofw2ukuasnwfdugssgoj63whgqswy/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-5.1.0-hn2mdexwfceriurodwmlvpow2kudhzt2/include -isystem /gpfs/alpine/csc439
            /world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/zfp-0.5.5-z7htnpirt3x377cljq54qdnx4aphipaj/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/include -O2 -g -DNDEBUG -fPI
            C -Wall -Wno-overloaded-virtual -x hip --offload-arch=gfx900 --offload-arch=gfx906 --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx1030 -fopenmp -std=c++17 -
            MD -MT CMakeFiles/strumpack.dir/src/SparseSolverMPIDist.cpp.o -MF CMakeFiles/strumpack.dir/src/SparseSolverMPIDist.cpp.o.d -o CMakeFiles/strumpack.dir/src/SparseSolverMPID
            ist.cpp.o -c /tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src/SparseSolverMPIDist.cpp
     178    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/lib/spack/env/cce/case-insensitive/crayCC -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -Dstrump
            ack_EXPORTS -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-maste
            r-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-build-mp5kr4f -isystem /opt/rocm-5.2.0/include -isystem /opt/rocm-5.2.0/llvm/lib/clang/14.0.0/.. -isystem /gpfs/alpine/csc439/worl
            d-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qrtofw2ukuasnwfdugssgoj63whgqswy/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-5.1.0-hn2mdexwfceriurodwmlvpow2kudhzt2/include -isystem /gpfs/alpine/csc439
            /world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/zfp-0.5.5-z7htnpirt3x377cljq54qdnx4aphipaj/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/include -O2 -g -DNDEBUG -fPI
            C -Wall -Wno-overloaded-virtual -x hip --offload-arch=gfx900 --offload-arch=gfx906 --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx1030 -fopenmp -std=c++17 -
            MD -MT CMakeFiles/strumpack.dir/src/misc/Triplet.cpp.o -MF CMakeFiles/strumpack.dir/src/misc/Triplet.cpp.o.d -o CMakeFiles/strumpack.dir/src/misc/Triplet.cpp.o -c /tmp/lpe
            yrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src/misc/Triplet.cpp
     179    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/lib/spack/env/cce/case-insensitive/crayCC -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -Dstrump
            ack_EXPORTS -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src -I/tmp/lpeyrala/spack-stage/spack-stage-strumpack-maste
            r-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-build-mp5kr4f -isystem /opt/rocm-5.2.0/include -isystem /opt/rocm-5.2.0/llvm/lib/clang/14.0.0/.. -isystem /gpfs/alpine/csc439/worl
            d-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/parmetis-4.0.3-qrtofw2ukuasnwfdugssgoj63whgqswy/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/metis-5.1.0-hn2mdexwfceriurodwmlvpow2kudhzt2/include -isystem /gpfs/alpine/csc439
            /world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/zfp-0.5.5-z7htnpirt3x377cljq54qdnx4aphipaj/include -isystem /gpfs/alpine/csc439/
            world-shared/E4S/ParaTools/22.11/PrgEnv-cray/spack/opt/spack/linux-sles15-zen3/cce-14.0.2/butterflypack-2.2.2-ndl4nqi4ug5dej5oawllsmmwe3dpaana/include -O2 -g -DNDEBUG -fPI
            C -Wall -Wno-overloaded-virtual -x hip --offload-arch=gfx900 --offload-arch=gfx906 --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx1030 -fopenmp -std=c++17 -
            MD -MT CMakeFiles/strumpack.dir/src/sparse/CompressedSparseMatrix.cpp.o -MF CMakeFiles/strumpack.dir/src/sparse/CompressedSparseMatrix.cpp.o.d -o CMakeFiles/strumpack.dir/
            src/sparse/CompressedSparseMatrix.cpp.o -c /tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-src/src/sparse/CompressedSparseMat
            rix.cpp
  >> 180    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 181    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 182    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 183    make[2]: *** [CMakeFiles/strumpack.dir/build.make:79: CMakeFiles/strumpack.dir/src/sparse/fronts/FrontalMatrixHIP.hip.cpp.o] Error 1
     184    make[2]: *** Waiting for unfinished jobs....
  >> 185    make[2]: *** [CMakeFiles/strumpack.dir/build.make:149: CMakeFiles/strumpack.dir/src/SparseSolver.cpp.o] Error 1
  >> 186    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 187    make[2]: *** [CMakeFiles/strumpack.dir/build.make:93: CMakeFiles/strumpack.dir/src/dense/HIPWrapper.hip.cpp.o] Error 1
  >> 188    make[2]: *** [CMakeFiles/strumpack.dir/build.make:121: CMakeFiles/strumpack.dir/src/StrumpackOptions.cpp.o] Error 1
  >> 189    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 190    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 191    clang-14clang-14: : clang-14: error: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 192    error:
     193    clang-14unsupported option '-fopenmp-targets=' for language mode 'HIP'
     194    : error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 195    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 196    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 197    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 198    make[2]: *** [CMakeFiles/strumpack.dir/build.make:261: CMakeFiles/strumpack.dir/src/sparse/CompressedSparseMatrix.cpp.o] Error 1
  >> 199    clang-14: error: unsupported option '-fopenmp-targets=' for language mode 'HIP'
  >> 200    make[2]: *** [CMakeFiles/strumpack.dir/build.make:107: CMakeFiles/strumpack.dir/src/StrumpackParameters.cpp.o] Error 1
  >> 201    make[2]: *** [CMakeFiles/strumpack.dir/build.make:191: CMakeFiles/strumpack.dir/src/SparseSolverMPIDist.cpp.o] Error 1
  >> 202    make[2]: *** [CMakeFiles/strumpack.dir/build.make:205: CMakeFiles/strumpack.dir/src/SparseSolverMixedPrecisionMPIDist.cpp.o] Error 1
  >> 203    make[2]: *** [CMakeFiles/strumpack.dir/build.make:219: CMakeFiles/strumpack.dir/src/misc/TaskTimer.cpp.o] Error 1
  >> 204    make[2]: *** [CMakeFiles/strumpack.dir/build.make:247: CMakeFiles/strumpack.dir/src/sparse/MC64ad.cpp.o] Error 1
  >> 205    make[2]: *** [CMakeFiles/strumpack.dir/build.make:163: CMakeFiles/strumpack.dir/src/SparseSolverMixedPrecision.cpp.o] Error 1
  >> 206    make[2]: *** [CMakeFiles/strumpack.dir/build.make:233: CMakeFiles/strumpack.dir/src/misc/Triplet.cpp.o] Error 1
  >> 207    make[2]: *** [CMakeFiles/strumpack.dir/build.make:177: CMakeFiles/strumpack.dir/src/StrumpackSparseSolverC.cpp.o] Error 1
  >> 208    make[2]: *** [CMakeFiles/strumpack.dir/build.make:135: CMakeFiles/strumpack.dir/src/SparseSolverBase.cpp.o] Error 1
     209    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-build-mp5kr4f'
  >> 210    make[1]: *** [CMakeFiles/Makefile2:536: CMakeFiles/strumpack.dir/all] Error 2
     211    make[1]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-strumpack-master-mp5kr4fsx632bllit2gmrxdazmexngxs/spack-build-mp5kr4f'
  >> 212    make: *** [Makefile:139: all] Error 2
```

Compiler definition is here:
```
  compilers:
  - compiler:
      spec: cce@14.0.2
      paths:
        cc: cc
        cxx: CC
        f77: ftn
        fc: ftn
      flags: {}
      operating_system: sles15
      target: any
      modules:
      - PrgEnv-cray/8.3.3
      - cce/14.0.2
      - xpmem
      - craype-x86-trento
      - libfabric
      - cray-pmi/6.1.3
      environment:
        set:
          RFE_811452_DISABLE: '1'
        append_path:
          PKG_CONFIG_PATH: /opt/cray/xpmem/2.4.4-2.3_11.2__gff0e1d9.shasta/lib64/pkgconfig
          LD_LIBRARY_PATH: /opt/cray/pe/gcc-libs:/opt/cray/libfabric/1.15.0.0/lib64
        prepend_path:
          LD_LIBRARY_PATH: /opt/cray/pe/pmi/6.1.3/lib
      extra_rpaths:
      - /opt/cray/pe/gcc-libs
      - /opt/cray/gcc-libs
```

HIP external defined like this:
```
    hip:
      buildable: false
      externals:
      - spec: hip@5.2.0
        prefix: /opt/rocm-5.2.0
        modules:
        - craype-accel-amd-gfx90a
        - rocm/5.2.0
        extra_attributes:
          compilers:
            c: CC
            c++: CC
            hip: CC
        environment:
          set:
            MPICH_GPU_SUPPORT_ENABLED: 1
```

We should be able to use `hipcc` via CC according to:
* https://docs.olcf.ornl.gov/systems/crusher_quick_start_guide.html#hip

But also, all of these other packages build in the same environment, same compiler def for `cce@14.0.2` and same set of ROCm 5.2.0 externals:
```
  - amrex +rocm amdgpu_target=gfx90a
  - arborx +rocm amdgpu_target=gfx90a
  - caliper +rocm amdgpu_target=gfx90a
  - chai ~benchmarks +rocm amdgpu_target=gfx90a
  - ecp-data-vis-sdk@1.0 +vtkm +rocm amdgpu_target=gfx90a # required vtk-m patch from ryan
  - gasnet +rocm amdgpu_target=gfx90a
  - ginkgo +rocm amdgpu_target=gfx90a
  - heffte +rocm amdgpu_target=gfx90a
  - hpx max_cpu_count=512 +rocm amdgpu_target=gfx90a
  - hypre +rocm amdgpu_target=gfx90a
  - kokkos +rocm amdgpu_target=gfx90a
  - magma ~cuda +rocm amdgpu_target=gfx90a
  - mfem +rocm amdgpu_target=gfx90a  # required adding explicit depends_on for rocsparse
  - papi +rocm amdgpu_target=gfx90a
  - petsc +rocm amdgpu_target=gfx90a
  - raja ~openmp +rocm amdgpu_target=gfx90a
  - slate +rocm amdgpu_target=gfx90a
  - sundials +rocm amdgpu_target=gfx90a
  - superlu-dist +rocm amdgpu_target=gfx90a
  - tasmanian ~openmp +rocm amdgpu_target=gfx90a
  - trilinos@13.4.0 +amesos +amesos2 +anasazi +aztec ~belos +boost +epetra +epetraext
    +ifpack ~ifpack2 +intrepid +intrepid2 +isorropia +kokkos +ml +minitensor +muelu
    +nox +piro +phalanx +rol +rythmos +sacado +stk +shards +shylu ~stokhos +stratimikos
    +teko +tempus +tpetra +trilinoscouplings +zoltan +zoltan2 +superlu-dist gotype=long_long
    +rocm amdgpu_target=gfx90a
  - umpire +rocm amdgpu_target=gfx90a
  - upcxx +rocm amdgpu_target=gfx90a
```

What do you think?

--- a/var/spack/repos/builtin/packages/strumpack/package.py
+++ b/var/spack/repos/builtin/packages/strumpack/package.py
@@ -146,6 +146,7 @@ class Strumpack(CMakePackage, CudaPackage, ROCmPackage):
                 args.append("-DCUDA_NVCC_FLAGS={0}".format(" ".join(self.cuda_flags(cuda_archs))))
 
         if "+rocm" in spec:
+            args.append("-DCMAKE_CXX_COMPILER={0}".format(spec["hip"].hipcc))
             args.append("-DHIP_ROOT_DIR={0}".format(spec["hip"].prefix))
             rocm_archs = spec.variants["amdgpu_target"].value
             hipcc_flags = []

--- a/var/spack/repos/builtin/packages/strumpack/package.py
+++ b/var/spack/repos/builtin/packages/strumpack/package.py
@@ -98,6 +98,9 @@ class Strumpack(CMakePackage, CudaPackage, ROCmPackage):
     patch("intel-19-compile.patch", when="@3.1.1")
     patch("shared-rocm.patch", when="@5.1.1")
 
+    # https://github.com/pghysels/STRUMPACK/commit/e4b110b2d823c51a90575b77ec1531c699097a9f
+    patch("strumpack-7.0.1-mpich-hipcc.patch", when="@7.0.1 +rocm ^mpich")
+
     def cmake_args(self):
         spec = self.spec
 

--- a/var/spack/repos/builtin/packages/strumpack/strumpack-7.0.1-mpich-hipcc.patch
+++ b/var/spack/repos/builtin/packages/strumpack/strumpack-7.0.1-mpich-hipcc.patch
@@ -1,0 +1,26 @@
+diff -ruN spack-src/CMakeLists.txt spack-src-patched/CMakeLists.txt
+--- spack-src/CMakeLists.txt	2022-10-12 11:48:07.000000000 -0700
++++ spack-src-patched/CMakeLists.txt	2023-01-30 13:35:11.271496106 -0800
+@@ -586,7 +586,8 @@
+ 
+ if(STRUMPACK_USE_MPI)
+   target_link_libraries(strumpack PUBLIC
+-    MPI::MPI_CXX MPI::MPI_C MPI::MPI_Fortran)
++    MPI::MPI_CXX MPI::MPI_C ${MPI_Fortran_LIBRARIES})
++  ## adding the MPI::MPI_Fortran target risks adding unwanted compile flags
+ endif()
+ 
+ if(OpenMP_FOUND)
+@@ -654,11 +655,7 @@
+     ButterflyPACK::sbutterflypack
+     ButterflyPACK::dbutterflypack
+     ButterflyPACK::cbutterflypack
+-    ButterflyPACK::zbutterflypack
+-    MPI::MPI_Fortran ${MPI_Fortran_LIBRARIES})
+-  ## for some reason just adding the MPI::MPI_Fortran target only
+-  ## links with the f08 module, not with the regular fortran MPI
+-  ## library
++    ButterflyPACK::zbutterflypack)
+ endif()
+ 
+ if(slate_FOUND)

